Swarmkit cannot be built with golang 1.6.3

Error
---
```
michel@homebox ~/go/src/github.com/docker/swarmkit $ make binaries
🐳 bin/swarmd
# github.com/docker/swarmkit/vendor/github.com/docker/docker/client/transport
vendor/github.com/docker/docker/client/transport/tlsconfig_clone.go:10: c.Clone undefined (type *tls.Config has no field or method Clone)
# github.com/docker/swarmkit/api
api/dispatcher.pb.go:1574: p.cluster undefined (type *raftProxyDispatcherServer has no field or method cluster)
api/dispatcher.pb.go:1581: p.connSelector.Conn undefined (type raftselector.ConnProvider has no field or method Conn)
api/dispatcher.pb.go:1593: p.connSelector.Reset undefined (type raftselector.ConnProvider has no field or method Reset)
Makefile:89: recipe for target 'bin/swarmd' failed
make: *** [bin/swarmd] Error 2
```

Go Version
---
```
michel@homebox ~/go/src/github.com/docker/swarmkit $ go version
go version go1.6.3 linux/amd64
```

Clone bug is fixed with revendor the docker/docker/client/transport .
now here is another error:

```
# github.com/docker/swarmkit/api
api/dispatcher.pb.go:1574: p.cluster undefined (type *raftProxyDispatcherServer has no field or method cluster)
api/dispatcher.pb.go:1581: p.connSelector.Conn undefined (type raftselector.ConnProvider has no field or method Conn)
api/dispatcher.pb.go:1593: p.connSelector.Reset undefined (type raftselector.ConnProvider has no field or method Reset)
make: *** [bin/swarmd] Error 2
```

--- a/README.md
+++ b/README.md
@@ -80,6 +80,7 @@ Requirements:
 
 -   Go 1.6 or higher
 -   A [working golang](https://golang.org/doc/code.html) environment
+-   [Protobuf 3.x or higher] (https://developers.google.com/protocol-buffers/docs/downloads) to regenerate protocol buffer files (e.g. using `make generate`)
 
 *SwarmKit* is built in Go and leverages a standard project structure to work well with Go tooling.
 If you are new to Go, please see [BUILDING.md](BUILDING.md) for a more detailed guide.

